### PR TITLE
Consolidate IsExternalInit definitions

### PR DESCRIPTION
**Pull request by tmat**
Remove redundant definitions and link the one defined in compilers instead.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler @dotnet/roslyn-ide

**Review by chsienki — APPROVED** [review]

File: src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/Microsoft.CommonLanguageServerProtocol.Framework.csproj [hunk 1]
```diff
@@ -16,6 +16,10 @@
     <None Include="License.txt" PackagePath="" Pack="true" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Include="..\..\..\Compilers\Core\Portable\InternalUtilities\IsExternalInit.cs" Link="Utilities\IsExternalInit.cs" />
+  </ItemGroup>
+
   <ItemGroup>
```
Comment (jcouv) [inline src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/Microsoft.CommonLanguageServerProtocol.Framework.csproj:23]: nit: unnecessary empty line above

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1) with a nit to consider

File: src/Compilers/Core/Portable/InternalUtilities/CompilerFeatureRequiredAttribute.cs [hunk 2]
```diff
@@ -5,6 +5,8 @@
 // Copied from:
 // https://github.com/dotnet/runtime/blob/fdd104ec5e1d0d2aa24a6723995a98d0124f724b/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/CompilerFeatureRequiredAttribute.cs
 
+#if !NET6_0_OR_GREATER
```
Comment (sharwell) [inline src/Compilers/Core/Portable/InternalUtilities/CompilerFeatureRequiredAttribute.cs:8]: 💡 You'll probably want a type forwarder here so the API binding surface is the same in netstandard2.0 and net6.0

**Pull request merged**